File: 0929-unique-email-addresses/0929-unique-email-addresses.cpp
class Solution {
public:
    int numUniqueEmails(vector<string>& emails) {
        set<string> s;
        for (string e : emails) {
            for (int i = 0; i < e.size(); ++i) {

                if (e[i] == '.') e.erase(i, 1);

                else if (e[i] == '+') {
                    while (e[i] != '@') e.erase(i, 1);
                };

                if (e[i] == '@') {
                    s.insert(e);
                    break;
                };
            };
        };
        return s.size();
    };
};

File: 0929-unique-email-addresses/0929-unique-email-addresses.py
class Solution:
    def numUniqueEmails(self, emails: List[str]) -> int:
        emailSet = set()
        for i in range(0, len(emails)):
            plusFlag = False
            j=0

            while j < len(emails[i]):
                if emails[i][j] == '@': 
                    emailSet.add(emails[i])
                    break
                    
                if emails[i][j] == '+': plusFlag = True
                if plusFlag == True:
                    emails[i] = emails[i][:j] + emails[i][(j+1):]
                    j-=1

                if emails[i][j] == '.': 
                    emails[i] = emails[i][:j] + emails[i][(j+1):]
                    j-=1

                j+=1

        return len(emailSet)